### ISSUE TITLE
Unquote doesn't work with Glob

### ISSUE THREAD

**Issue by ngeiswei** [opencog/atomspace#1015]
Given

```
(And (Concept "A") (Concept "B") (Concept "C"))
```

The following works, that is fetches a list of A, B and C, in some order.

```
(cog-bind (Bind (TypedVariable (Glob "$X") (Type "ConceptNode")) (LocalQuote (And (Glob "$X"))) (Glob "$X")))
```

while the follow doesn't

```
(cog-bind (Bind (TypedVariable (Glob "$X") (Type "ConceptNode")) (Quote (And (Unquote (Glob "$X")))) (Glob "$X")))
```

**Comment by linas** [comment]
Heh.  Do you want to fix this or should I?  I have about 20 minutes to look at this tonight.  I was hoping that this would work, but it doesn't:
```
(define (bc)
   (cog-bind (Bind (TypedVariable (Glob "$X") (Type "ConceptNode")) 
       (Present (Quote (And (Unquote (Glob "$X")))) )
       (Glob "$X"))))
```

**Comment by ngeiswei** [comment]
@linas I don't think I need it ATM, and when I'll do I can look into it.

**Comment by linas** [comment]
Heh.  Do you want to fix this or should I?  I have about 20 minutes to look at this tonight.  I was hoping that this would work, but it doesn't:
```
(define (bc)
   (cog-bind (Bind (TypedVariable (Glob "$X") (Type "ConceptNode")) 
       (Present (Quote (And (Unquote (Glob "$X")))) )
       (Glob "$X"))))
```

**Comment by linas** [comment]
Heh.  Do you want to fix this or should I?  I have about 20 minutes to look at this tonight.  I was hoping that this would work, but it doesn't:
```
(define (bc)
   (cog-bind (Bind (TypedVariable (Glob "$X") (Type "ConceptNode")) 
       (Present (Quote (And (Unquote (Glob "$X")))) )
       (Glob "$X"))))
```

**Comment by linas** [comment]
OK, I poked around a bit, and it seems like the infrastructure for quoting anything except variables is missing. If this is not urgent then I don't want to mess with this right now; the computer I'm working on this week is too slow for this style of programming.

**Comment by linas** [comment]
closing; this got fixed by @kasimebrahim  in pull req #2393
